lib/fontist/font.rb

```diff
@@ -50,6 +56,58 @@ def check_or_create_fontist_path!
       end
     end
 
+    def missing_fonts(existing_paths)
+      available = available_fonts || []
+      existing = filenames(existing_paths)
+      available - existing
+    end
+
+    def available_fonts
+      available_by_style || available_by_font || available_by_formula
+    end
+
+    def available_by_style
+      return unless @style
+
+      if formula
+        formula.fonts.select do |f|
+          if f.name.casecmp?(@name)
+            f.styles.select do |s|
+              s.font if s.type.casecmp?(@style)
+            end
+          end
+        end.flatten
+      end
+    end
+
+    def available_by_font
+      if formula
+        formula.fonts.select do |f|
+          if f.name.casecmp?(@name)
+            f.styles.map do |s|
+              s.font
+            end
+          end
+        end.flatten
+      end
+    end
+
+    def available_by_formula
+      if formula
```

Comment:
How about shortcutting it with `return unless formula`?

Reply:
Highlighted your comment in a new PR

https://github.com/fontist/fontist/pull/149#discussion_r498748029